src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -79,7 +82,13 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            hasError = true;
          }
+
+         if(hasError){
+            response.sendRedirect(errorURL);
```

Comment:
Ignore this default thing.  The default 404 is in place for the root level, not how the individual apps handle 404's.  My bad.